src/riak_pb_codec.erl

```diff
@@ -95,66 +95,49 @@ encode(Msg) when is_tuple(Msg) ->
 
 %% ------------------------------------------------------------
 %% Encode a message when no content body is present (message atom
-%% only).  
-%% 
+%% only).
+%%
 %% For PB messages, this simply encodes the message code, which serves
 %% to identify the encoded message on the other side of the socket
 %% connection.
-%%
-%% For TTB messages, we cannot do this, since all TTB messages have
-%% the same message code.  Instead, we encode a message consisting of
-%% the atom identifying the message type
 %% ------------------------------------------------------------
 
-encode_msg_no_body(?TTB_MSG_CODE, Msg) ->
-    encode({Msg, <<>>});
 encode_msg_no_body(MsgCode, _Msg) ->
     [MsgCode]. %% I/O layer will convert this to binary
 
+%% @doc Convert a property list to an RpbBucketProps message
+%% @private
+post_decode(Msg=#tsgetreq{key=K}) ->
+    Msg#tsgetreq{key=riak_pb_ts_codec:decode_cells(K)};
+post_decode(Msg=#tsputreq{rows=R}) ->
+    Msg#tsputreq{rows=riak_pb_ts_codec:decode_rows(R)};
+post_decode(Msg) ->
+    Msg.
+
 %% @doc Decode a protocol buffer message given its type - if no bytes
 %% return the atom for the message code. Replaces `riakc_pb:decode/2'.
-
 -spec decode(integer(), binary()) -> atom() | tuple().
+decode(MsgCode, <<>>) ->
+    msg_type(MsgCode);
 decode(MsgCode, MsgData) ->
     Decoder = decoder_for(MsgCode),
-    Decoder:decode(msg_type(MsgCode), MsgData).
+    Decoded = Decoder:decode(msg_type(MsgCode), MsgData),
+    post_decode(Decoded).
```

Comment:
This is now necessary as some TS PB specific decoding that used to be in the `riak_kv` service is now here. As other messages are converted to be PB / TTB any PB-specific encoding / decoding should be moved to `riak_pb`